schema_based_on_diagram.sql

```diff
@@ -0,0 +1,37 @@
+DROP DATABASE IF EXISTS clinic;
+CREATE DATABASE clinic;
+
+CREATE TABLE medical_histories(
+  id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  admitted_at timestamp,
+  patient_id INTEGER,
+  FOREIGN KEY (patient_id) REFERENCES patients(id)
+);
+
+CREATE TABLE invoices(
+  id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  total_amount decimal,
+  generated_at timestamp,
+  payed_at timestamp,
+  FOREIGN KEY (medical_histories_id) REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoice_items(
+  id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  unit_price decimal,
+  quantity integer,
+  total_price decimal,
+  FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+  FOREIGN KEY (treatment_id) REFERENCES treatment(id)
+);
+
+CREATE TABLE treatments(
+  id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  type varchar(255),
+  name varchar(255),
+);
+
```

Comment:
- Please add a join table for the many-to-many relationship between medical_histories and the treatments tables. This wasn't mentioned in the requirements but it's going to help you avoid problems along the way.
- The table can look something like;
- ```ruby
  CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```
Note: don't forget to add indexes for the new table.